failed to build v22.2.8 from source

**Describe the problem**

I'm trying to build v22.2.8 from source with "make" (which I know is deprecated, but I believe it's still supported?).  I'm getting this error:

```
src/transactionDetails/transactionDetails.stories.tsx(34,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: { stats_data: { statement_fingerprint_ids: Long[]; app: string; stats: { count: Long; max_retries: Long; ... 7 more ...; rows_written: { ...; }; }; aggregated_ts: Timestamp; transaction_fingerprint_id: Long; aggregation_interval: { ...; }; }; }; ...' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(52,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: null; isLoading: true; statements: undefined; nodeRegions: { "1": string; "2": string; "3": string; "4": string; }; isTenant: false; ... 8 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(70,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: undefined; transaction: undefined; isLoading: false; statements: undefined; nodeRegions: { "1": string; "2": string; "3": string; "4": string; }; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(90,8): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: undefined; isLoading: false; statements: ICollectedStatementStatistics[]; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionsPage/transactionsPage.stories.tsx(38,6): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; data: IStatementsResponse; timeScale: TimeScale; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; onFilterChange: (...args: any[]) => void; ... 11 more ...; isDataValid: true; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(60,8): error TS2741: Property 'isReqInFlight' is missing in type '{ isDataValid: true; columns: string[]; data: IStatementsResponse & { transactions: any[]; statements: any[]; }; timeScale: TimeScale; filters: Filters; ... 13 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(89,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: IStatementsResponse & { transactions: any[]; statements: any[]; }; timeScale: TimeScale; filters: Filters; ... 13 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(113,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: undefined; timeScale: TimeScale; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; onFilterChange: (...args: any[]) => void; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(136,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: undefined; timeScale: TimeScale; error: RequestError; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; ... 11 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build:typescript" exited with 2.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:370: pkg/ui/cluster-ui/dist/main.js] Error 1
```

**To Reproduce**

Here's roughly what I did: I stared with an Ubuntu 22.04 LTS instance in AWS.  (I used ami-0fcf52bcf5db7b003, instance type c5.4xlarge, with a 1024GiB gpt2 root volume.)

```
$ sudo apt-get update
$ sudo apt-get install libncurses-dev build-essential
$ sudo apt-get install nodejs
$ sudo apt-get install npm 
$ sudo apt-get install yacc 
$ sudo npm install -g yarn
$ sudo apt-get install autoconf
$ sudo apt-get install cmake
$ mkdir install
$ cd install
$ curl -LO https://go.dev/dl/go1.19.8.linux-amd64.tar.gz
$ tar xzf ../go1.19.8.linux-amd64.tar.gz
$ export GOPATH=$HOME/go
$ mkdir -p $(go env GOPATH)/src/github.com/cockroachdb
$ cd $(go env GOPATH)/src/github.com/cockroachdb
$ git clone https://github.com/cockroachdb/cockroach.git
$ cd cockroach
$ git checkout v22.2.8
$ make buildoss BUILDTYPE=release
```

I originally saw this on illumos, not Linux, and with some local patches.  Here I've reproduced it with stock 22.2.8 on Linux.  I see this with both the normal build (`make build BUILDTYPE=release`) and OSS build (see above) on both illumos and Linux.

Full build output: https://gist.github.com/davepacheco/86fe609d016a804a303240f155c9243c

Jira issue: CRDB-27171

--- a/Makefile
+++ b/Makefile
@@ -328,7 +328,7 @@ endif
 
 # Force vendor directory to rebuild.
 .PHONY: vendor_rebuild
-vendor_rebuild: bin/.submodules-initialized
+vendor_rebuild: bin/.submodules-initialized | fake-protobufs
 	$(GO_INSTALL) -v -mod=mod github.com/goware/modvendor
 	./build/vendor_rebuild.sh
 
@@ -367,6 +367,7 @@ CLUSTER_UI_JS := pkg/ui/cluster-ui/dist/main.js
 
 .SECONDARY: $(CLUSTER_UI_JS)
 $(CLUSTER_UI_JS): $(shell find pkg/ui/workspaces/cluster-ui/src -type f | sed 's/ /\\ /g') pkg/ui/yarn.installed pkg/ui/workspaces/db-console/src/js/protos.d.ts | bin/.submodules-initialized
+	$(NODE_RUN) -C pkg/ui/workspaces/cluster-ui yarn install --ignore-optional --offline
 	$(NODE_RUN) -C pkg/ui/workspaces/cluster-ui yarn build
 
 .SECONDARY: pkg/ui/yarn.installed
@@ -378,7 +379,7 @@ pkg/ui/yarn.installed: pkg/ui/package.json pkg/ui/yarn.lock | bin/.submodules-in
 	$(NODE_RUN) -C pkg/ui yarn install --ignore-optional --offline
 	touch $@
 
-vendor/modules.txt: | bin/.submodules-initialized
+vendor/modules.txt: | bin/.submodules-initialized fake-protobufs
 
 # Update the git hooks and install commands from dependencies whenever they
 # change.
@@ -793,6 +794,7 @@ SQLPARSER_TARGETS = \
 	pkg/sql/lexbase/reserved_keywords.go
 
 PROTOBUF_TARGETS := bin/.go_protobuf_sources bin/.gw_protobuf_sources
+$(PROTOBUF_TARGETS): fake-protobufs
 
 SWAGGER_TARGETS :=
   #docs/generated/swagger/spec.json
@@ -1248,6 +1250,12 @@ protobuf: $(PROTOBUF_TARGETS)
 protobuf: ## Regenerate generated code for protobuf definitions.
 	$(info $(yellow)[WARNING] Use `dev generate protobuf` instead.$(term-reset))
 
+.PHONY: fake-protobufs
+fake-protobufs:
+	set -e; for dir in $(sort $(dir $(GO_PROTOS))); do \
+	    echo "package $$(basename $$dir)" > $$dir/empty.pb.go; \
+	done
+
 # pre-push locally runs most of the checks CI will run. Notably, it doesn't run
 # the acceptance tests.
 .PHONY: pre-push
@@ -1329,6 +1337,7 @@ PROTO_MAPPINGS := $(PROTO_MAPPINGS)Mgoogle/api/annotations.proto=google.golang.o
 PROTO_MAPPINGS := $(PROTO_MAPPINGS)Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,
 PROTO_MAPPINGS := $(PROTO_MAPPINGS)Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,
 PROTO_MAPPINGS := $(PROTO_MAPPINGS)Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types,
+PROTO_MAPPINGS := $(PROTO_MAPPINGS)Metcd/raft/v3/raftpb/raft.proto=go.etcd.io/etcd/raft/v3/raftpb,
 
 GW_SERVER_PROTOS := ./pkg/server/serverpb/admin.proto ./pkg/server/serverpb/status.proto ./pkg/server/serverpb/authentication.proto
 GW_TS_PROTOS := ./pkg/ts/tspb/timeseries.proto
@@ -1852,7 +1861,7 @@ $(has-build-info): override LINKFLAGS += \
 	-X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=$(TARGET_TRIPLE)" \
 	$(if $(BUILDCHANNEL),-X "github.com/cockroachdb/cockroach/pkg/build.channel=$(BUILDCHANNEL)")
 
-$(bins): bin/%: bin/%.d | bin/prereqs bin/.submodules-initialized
+$(bins): bin/%: bin/%.d | bin/prereqs bin/.submodules-initialized fake-protobufs
 	@echo go install -v $*
 	$(PREREQS) $(if $($*-package),$($*-package),./pkg/cmd/$*) > $@.d.tmp
 	mv -f $@.d.tmp $@.d

--- a/pkg/cmd/protoc-gen-gogoroach/main.go
+++ b/pkg/cmd/protoc-gen-gogoroach/main.go
@@ -35,6 +35,7 @@ func fixImports(s string) string {
 		}
 
 		line = strings.ReplaceAll(line, "github.com/cockroachdb/cockroach/pkg/etcd", "go.etcd.io/etcd")
+		line = strings.ReplaceAll(line, "github.com/cockroachdb/cockroach/pkg/go.etcd.io", "go.etcd.io")
 		line = strings.ReplaceAll(line, "github.com/cockroachdb/cockroach/pkg/errorspb", "github.com/cockroachdb/errors/errorspb")
 		line = strings.ReplaceAll(line, "golang.org/x/net/context", "context")
 		if builtinRegex.MatchString(line) {

--- a/pkg/ui/workspaces/cluster-ui/tsconfig.json
+++ b/pkg/ui/workspaces/cluster-ui/tsconfig.json
@@ -25,6 +25,9 @@
   "exclude": [
     "node_modules",
     "dist",
+    "**/*.spec.ts",
+    "**/*.spec.tsx",
+    "**/*.stories.tsx",
     "**/*.test.*",
     ".jest"
   ]

--- a/pkg/ui/workspaces/db-console/tsconfig.json
+++ b/pkg/ui/workspaces/db-console/tsconfig.json
@@ -40,6 +40,9 @@
   },
   "exclude": [
     "cluster-ui",
+    "**/*.spec.ts",
+    "**/*.spec.tsx",
+    "**/*.stories.tsx",
     "dist"
   ]
 }

